src/prototype_creep_resources.js

```diff
@@ -694,14 +686,14 @@ let checkSourcerMatch = function(sourcers, source_id) {
   return false;
 };
 
-let checkSourcer = function(creep) {
-  const sources = creep.room.find(FIND_SOURCES);
-  const sourcers = creep.room.findPropertyFilter(FIND_MY_CREEPS, 'memory.role', ['sourcer']);
+Creep.prototype.checkSourcer = function() {
```

Comment:
This should be definitely as `Room.prototype` and search for the closest room. These checks should happen in the room and not on a creep.